requirements-extras.txt

```
@@ -10,7 +10,7 @@ notebook>=6.0.0
 ipykernel>=6.21.2
 ghp-import>=1.0.0
 aiohttp>=3.8.6
-watchdog>=0.8.3
+watchdog>=4.0.0
```

It seems that we could allow older versions as well. Arch uses watchdog 2.3.0, and [it has the `src_path` property](https://github.com/gorakhargosh/watchdog/blob/4f83d701f05c446c909fec79b44a7a87062b7378/src/watchdog/events.py#L136-L139), so let’s make this the minimum version instead?


```suggestion
watchdog>=2.3.0
```

---

Actually you probably can even keep `watchdog>=0.8.3`, since it seems to have been there all along: https://github.com/gorakhargosh/watchdog/commit/ef2f75a611e4c9323bd69e91bc8b2152cf252683#diff-24fdae1923051e85658e62d7003f4484086b26f93d53b6c9263639f018846301R55

---

We have been using access to the attribute `_src` in our `auto` plugin. One should not access a underscore-named attribute in other people's software, so our code has always been a bit dubious.

That attribute is gone with 4.0.0 and 4.0.1. See #3761 for details. But a new official (API) attribute `src` has come in its place. This is what is being used here.

I have not investigated whether that attribute `src` is present in older versions and don't want to spend my time on that in the next few days. If you want to spend your time on that investigation, do. Afaik, any version that has the `src` attribute will work with Nikola.

---

Looks good to me at first glance. I'm busy the next days - someone change the version number and merge to `master`.